src/util/xml.h

```diff
@@ -135,4 +137,20 @@ inline std::ostream &operator<<(
   return out;
 }
 
+bool operator==(const xmlt &a, const xmlt &b);
+bool operator!=(const xmlt &a, const xmlt &b);
+
+/// Convert the structured_data into an xml object. For the example structured
```

Comment:
As with the json version, `structured_data` -> `structured_datat` and the example structure data will likely need copying into this comment as well. (sorry again...)